File: frontend/src/types/task.ts
export interface Task {
  id: string;
  message_id: string;
  conversation_id: string;
  agent_id: string;
  sequence: number;
  name: string;
  status: 'pending' | 'running' | 'completed' | 'failed';
  result?: string | null;
  error_message?: string | null;
  started_at?: string | null;
  completed_at?: string | null;
  created_at: string;
  agent_name?: string | null;
}

export interface TaskUpdateEvent {
  task_id: string;
  sequence: number;
  name: string;
  status: string;
  result?: string | null;
  error_message?: string | null;
  agent_id?: string | null;
  agent_name?: string | null;
}

File: backend/app/schemas/task.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class TaskResponse(BaseModel):
    id: str
    message_id: str
    conversation_id: str
    agent_id: str
    sequence: int
    name: str
    status: str
    result: Optional[str] = None
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class TaskUpdateEvent(BaseModel):
    task_id: str
    sequence: int
    name: str
    status: str
    result: Optional[str] = None
    error_message: Optional[str] = None
    agent_id: Optional[str] = None
    agent_name: Optional[str] = None
